backend/src/nodes/onnx_nodes.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+import os
+from typing import Tuple
+
+import numpy as np
+import onnx
+import onnxruntime as ort
+from sanic.log import logger
+
+from .categories import ONNX
+from .node_base import NodeBase
+from .node_factory import NodeFactory
+from .properties.inputs import *
+from .properties.outputs import *
+from .utils.onnx_auto_split import onnx_auto_split_process
+from .utils.utils import get_h_w_c, np2nptensor, nptensor2np
+
+
+@NodeFactory.register("chainner:onnx:load_model")
+class OnnxLoadModelNode(NodeBase):
+    """ONNX Load Model node"""
+
+    def __init__(self):
+        """Constructor"""
+        super().__init__()
+        self.description = (
+            """Load ONNX model file (.onnx). Theoretically supports any ONNX model."""
+        )
+        self.inputs = [OnnxFileInput()]
+        self.outputs = [OnnxModelOutput(), TextOutput("Model Name")]
+
+        self.category = ONNX
+        self.name = "Load Model"
+        self.icon = "ONNX"
+        self.sub = "Input & Output"
+
+        self.model = None  # Defined in run
+
+    def run(self, path: str) -> Tuple[ort.InferenceSession, str]:
+        """Read a pth file from the specified path and return it as a state dict
+        and loaded model after finding arch config"""
+
+        assert os.path.exists(path), f"Model file at location {path} does not exist"
+
+        assert os.path.isfile(path), f"Path {path} is not a file"
+
+        logger.info(f"Reading onnx model from path: {path}")
+        model = onnx.load_model(path)
+
+        model_as_string = model.SerializeToString()  # type: ignore
+
+        session = ort.InferenceSession(
+            model_as_string,
+            providers=[
+                "CPUExecutionProvider"
+                if os.environ["device"] == "cpu"
+                else "CUDAExecutionProvider"
+            ],
+        )
+
+        basename = os.path.splitext(os.path.basename(path))[0]
+
+        return session, basename
+
+
+@NodeFactory.register("chainner:onnx:upscale_image")
+class OnnxImageUpscaleNode(NodeBase):
+    """Image Upscale node"""
+
+    def __init__(self):
+        """Constructor"""
+        super().__init__()
+        self.description = "Upscales an image using an ONNX Super-Resolution model. \
+            ONNX does not support automatic out-of-memory handling via automatic tiling. \
+            Therefore, you must set a split factor yourself. If you get an out-of-memory error, increase this number by 1."
+        self.inputs = [
+            OnnxModelInput(),
+            ImageInput(),
+            NumberInput("Split Factor", default=1, minimum=1, maximum=10),
```

Comment:
We could make it tile size. This was just easy to do since the logic is already there.

The problem with estimating a good tile size based on available vram is that the vram needed changes based on the model used. For example lite models use much less vram. There isn't a one-size-fits all solution, which is why I did away with tilesize altogether in favor of the auto-split algorithm. 

if we want to switch this to tilesize I'd rather it just be a manual setting, even though that kinda sucks.

Reply:
There is an overlap. There are no hard edges

Reply:
You might be confused because it doesn't fade the edges together, but it doesn't need to. All it needs is to upscale with the overlap and then chop the overlap off and it works fine

Reply:
Huh, as long as it works, it's fine I guess.

I still think we should use tile sizes though.

Reply:
I'm fine with using tile size for onnx, I'll just have to do more work to get that working and tested. Maybe instead we could do "fake" tile size in that we can just find the first split factor that makes the resulting tiles less than the amount specified. That would make it slightly easier. But I can also just implement a normal tile size algo.

Reply:
And by the way, the auto-split algorithm is automatic. It just tries to upscale and if it can't it splits the image. For both pytorch and ncnn this works well because it allocates all the vram it needs very quickly, so it fails quickly and therefore wastes hardly any time just attempting the whole thing first. So best case scenario you end up with a much much faster upscale from not splitting at all. 

You probably aren't thinking about it as auto because you think something that would guess the tile size would be better, but that would involve so many different factors that we simply can't account for. IEU tried to do something like that and it failed more than it worked. My method works every time, for any model

Reply:
> "Fake" tile size sounds good.

Sounds good. I'll try that out when I have time later